src/index.js

```diff
@@ -315,84 +310,79 @@ class ImageMinimizerPlugin {
       )
     ).flat();
 
-    const cpus = os.cpus() || { length: 1 };
-    const limit = this.options.concurrency || Math.max(1, cpus.length - 1);
-
+    // In some cases cpus() returns undefined
+    // https://github.com/nodejs/node/issues/19022
+    const limit = Math.max(
+      1,
+      this.options.concurrency ?? os.cpus()?.length ?? 1
+    );
     const { RawSource } = compiler.webpack.sources;
 
-    const scheduledTasks = [];
+    const scheduledTasks = assetsForTransformers.map((asset) => async () => {
```

Comment:
I am fine with it, but based on `jsperf`, `for ... of ...` loop is faster then `map`, try it locally

Reply:
not always. It seems dependent on array size. And in this case `map` should be faster https://jsben.ch/olwNK

Reply:
Looks good, seems V8 improves perf here